pkg/pod/script.go

```diff
@@ -116,117 +105,125 @@ func convertScripts(shellImageLinux string, shellImageWin string, steps []v1beta
+	for i, _ := range containers {
+		debugInfoVolumeMount := corev1.VolumeMount{
+			Name:      debugInfoVolumeName,
+			MountPath: filepath.Join(debugInfoDir, fmt.Sprintf("%d", i)),
 		}
-		debugScripts := []script{{
-			name:    "continue",
-			content: defaultScriptPreamble + fmt.Sprintf(debugContinueScriptTemplate, len(steps), debugInfoDir, RunDir),
-		}, {
-			name:    "fail-continue",
-			content: defaultScriptPreamble + fmt.Sprintf(debugFailScriptTemplate, len(steps), debugInfoDir, RunDir),
-		}}
+		(&containers[i]).VolumeMounts = append((&containers[i]).VolumeMounts, debugScriptsVolumeMount, debugInfoVolumeMount)
+	}
+	*placeScripts = true
 
-		// Add debug or breakpoint related scripts to /tekton/debug/scripts
-		// Iterate through the debugScripts and add routine for each of them in the initContainer for their creation
-		for _, debugScript := range debugScripts {
-			tmpFile := filepath.Join(debugScriptsDir, fmt.Sprintf("%s-%s", "debug", debugScript.name))
-			heredoc := names.SimpleNameGenerator.RestrictLengthWithRandomSuffix(fmt.Sprintf("%s-%s-heredoc-randomly-generated", "debug", debugScript.name))
+	type script struct {
+		name    string
+		content string
+	}
+	debugScripts := []script{{
+		name:    "continue",
+		content: defaultScriptPreamble + fmt.Sprintf(debugContinueScriptTemplate, len(containers), debugInfoDir, RunDir),
+	}, {
+		name:    "fail-continue",
+		content: defaultScriptPreamble + fmt.Sprintf(debugFailScriptTemplate, len(containers), debugInfoDir, RunDir),
+	}}
+
+	// Add debug or breakpoint related scripts to /tekton/debug/scripts
+	// Iterate through the debugScripts and add routine for each of them in the initContainer for their creation
+	for _, debugScript := range debugScripts {
+		tmpFile := filepath.Join(debugScriptsDir, fmt.Sprintf("%s-%s", "debug", debugScript.name))
+		heredoc := names.SimpleNameGenerator.RestrictLengthWithRandomSuffix(fmt.Sprintf("%s-%s-heredoc-randomly-generated", "debug", debugScript.name))
+
+		initContainer.Args[1] += fmt.Sprintf(initScriptDirective, tmpFile, heredoc, debugScript.content, heredoc)
+	}
+}
 
-			initContainer.Args[1] += fmt.Sprintf(initScriptDirective, tmpFile, heredoc, debugScript.content, heredoc)
-		}
+// convertListOfSidecars does the heavy lifting for convertScripts.
+//
+// It iterates through the list of sidecars, generates the script file name and heredoc termination string,
+// adds an entry to the init container args, sets up the step container to run the script, and sets the volume mounts.
+func convertListOfSidecars(sidecars []v1beta1.Sidecar, initContainer *corev1.Container, placeScripts *bool, namePrefix string) []corev1.Container {
+	containers := []corev1.Container{}
+	for i, s := range sidecars {
+		c := s.ToK8sContainer()
+		scriptFile := filepath.Join(scriptsDir, names.SimpleNameGenerator.RestrictLengthWithRandomSuffix(fmt.Sprintf("%s-%d", namePrefix, i)))
```

Comment:
can this line be a helper function? Also, is it now possible to have a file path conflict between step scripts and sidecar scripts?

Reply:
I'm not sure because this line relies on an parameter `i` that's part of the for loop, do you mean we can have a helper function that takes the file name and generates the filepath like the following?

```golang
scriptFile := getScriptFilePath(scriptDir, scriptName)
func getScriptFilePath(scriptDir, scriptName string) string{
  return filepath.Join(scriptDir, scriptName)
}
```

Reply:
maybe something like 

```
func getScriptFilePath(scriptDir, scriptName string) string{
  return filepath.Join(scriptsDir, names.SimpleNameGenerator.RestrictLengthWithRandomSuffix(scriptName))
}
```
?